diff -u -r VTK_original/Widgets/vtkImagePlaneWidget.cxx VTK_new/Widgets/vtkImagePlaneWidget.cxx
--- Widgets/vtkImagePlaneWidget.cxx	2013-02-25 17:02:17.154215200 +0100
+++ Widgets/vtkImagePlaneWidget.cxx	2013-02-25 17:16:37.097427300 +0100
@@ -2357,6 +2357,12 @@
 }
 
 //----------------------------------------------------------------------------
+void vtkImagePlaneWidget::GetTexturePlaneActor(vtkActor *act)
+{
+  act->ShallowCopy(this->TexturePlaneActor);
+}
+
+//----------------------------------------------------------------------------
 vtkPolyDataAlgorithm *vtkImagePlaneWidget::GetPolyDataAlgorithm()
 {
   return this->PlaneSource;
diff -u -r VTK_original/Widgets/vtkImagePlaneWidget.h VTK_new/Widgets/vtkImagePlaneWidget.h
--- Widgets/vtkImagePlaneWidget.h	2013-02-25 17:02:17.154215200 +0100
+++ Widgets/vtkImagePlaneWidget.h	2013-02-25 17:18:07.457553800 +0100
@@ -259,6 +259,9 @@
   // the points and polygons are added to it.
   void GetPolyData(vtkPolyData *pd);
 
+  // Added by Clinical Graphics
+  void GetTexturePlaneActor(vtkActor *act);
+
   // Description:
   // Satisfies superclass API.  This returns a pointer to the underlying
   // vtkPolyData.  Make changes to this before calling the initial PlaceWidget()
