Index: vtkProperty.h
===================================================================
RCS file: /cvsroot/VTK/VTK/Rendering/vtkProperty.h,v
retrieving revision 1.82
diff -u -r1.82 vtkProperty.h
--- vtkProperty.h	10 Mar 2009 15:51:25 -0000	1.82
+++ vtkProperty.h	18 Feb 2010 19:26:41 -0000
@@ -329,6 +329,104 @@
     this->AddShaderVariable(name, 3, v);
     }
 
+  // also need four element variables
+  void AddShaderVariable(const char* name, int v1, int v2, int v3, int v4)
+    {
+    int v[4];
+    v[0] = v1;
+    v[1] = v2;
+    v[2] = v3;
+    v[3] = v4;
+    this->AddShaderVariable(name, 4, v);
+    }
+  void AddShaderVariable(const char* name, float v1, float v2, float v3, float v4)
+    {
+    float v[4];
+    v[0] = v1;
+    v[1] = v2;
+    v[2] = v3;
+    v[3] = v4;
+    this->AddShaderVariable(name, 4, v);
+    }
+  void AddShaderVariable(const char* name, double v1, double v2, double v3, double v4)
+    {
+    double v[4];
+    v[0] = v1;
+    v[1] = v2;
+    v[2] = v3;
+    v[3] = v4;
+    this->AddShaderVariable(name, 4, v);
+    }
+
+
+
+
+
+  // Description:
+  // more methods for Python, it seems the wrapping can only
+  // distinguish overloaded methods by number of parameters and not by
+  // type, so in Python only the integer versions of the methods above
+  // are correctly wrapped.
+  void AddShaderVariableFloat(const char* name, float v1)
+  {
+    this->AddShaderVariable(name, v1);
+  }
+
+  void AddShaderVariableFloat(const char* name, float v1, float v2)
+  {
+    this->AddShaderVariable(name, v1, v2);
+  }
+  
+  void AddShaderVariableFloat(const char* name, float v1, float v2, float v3)
+  {
+    this->AddShaderVariable(name, v1, v2, v3);
+  }
+  
+  void AddShaderVariableFloat(const char* name, float v1, float v2, float v3, float v4)
+  {
+    this->AddShaderVariable(name, v1, v2, v3, v4);
+  }
+
+  void AddShaderVariableDouble(const char* name, double v1)
+  {
+    this->AddShaderVariable(name, v1);
+  }
+
+  void AddShaderVariableDouble(const char* name, double v1, double v2)
+  {
+    this->AddShaderVariable(name, v1, v2);
+  }
+  
+  void AddShaderVariableDouble(const char* name, double v1, double v2, double v3)
+  {
+    this->AddShaderVariable(name, v1, v2, v3);
+  }
+  
+  void AddShaderVariableDouble(const char* name, double v1, double v2, double v3, double v4)
+  {
+    this->AddShaderVariable(name, v1, v2, v3, v4);
+  }
+
+  void AddShaderVariableInt(const char* name, int v1)
+  {
+    this->AddShaderVariable(name, v1);
+  }
+
+  void AddShaderVariableInt(const char* name, int v1, int v2)
+  {
+    this->AddShaderVariable(name, v1, v2);
+  }
+  
+  void AddShaderVariableInt(const char* name, int v1, int v2, int v3)
+  {
+    this->AddShaderVariable(name, v1, v2, v3);
+  }
+  
+  void AddShaderVariableInt(const char* name, int v1, int v2, int v3, int v4)
+  {
+    this->AddShaderVariable(name, v1, v2, v3, v4);
+  }
+
   // Description:
   // Set/Get the texture object to control rendering texture maps.  This will
   // be a vtkTexture object. A property does not need to have an associated
